src/cargo/core/features.rs

```diff
@@ -839,6 +839,10 @@ pub struct GitoxideFeatures {
     /// Checkout git dependencies using `gitoxide` (submodules are still handled by git2 ATM, and filters
     /// like linefeed conversions are unsupported).
     pub checkout: bool,
+    /// A feature flag which doesn't have any meaning except for preventing
+    /// `CARGO_USE_GITOXIDE_INSTEAD_OF_GIT2=1` builds to enable all safe `gitoxide` features.
+    /// That way, `gitoxide` isn't actually used even though it's enabled.
+    pub internal_use_git2: bool,
```

Comment:
Since `-Zgitoxide` has higher precedence than `CARGO_USE_GITOXIDE_INSTEAD_OF_GIT2`, should we have a `-Zgitoxide=none` option to force the use of `git2` in the presence of `CARGO_USE_GITOXIDE_INSTEAD_OF_GIT2`?

This aligns with what we already have rather than adding additional feature flags to workaround our env hacks for testing another feature

Alternatively, could the tests set `env("CARGO_USE_GITOXIDE_INSTEAD_OF_GIT2", "0")`?  We'd need to update the `CARGO_USE_GITOXIDE_INSTEAD_OF_GIT2` checks from checking for variable presence to checking for either enabled or not disabled.

Reply:
I have implemented the alternative version in 775fc44702da2f8b6de3a1735d7297ae0a8f034a
 because it seemed easiest to allow the environment variables value to have a meaning. While at it, I remove the build-time dependency to the environment variable and made it runtime-evaluated instead. This means less builds on CI, faster running of tests locally and generally I don't know why I made it build-time evaluated in the first place.